### DISCUSSION TITLE
Crash on one machine: Plugin version Qt_5 not defined in file libQt5Qml.so.5 with link time reference

### DISCUSSION THREAD

**Discussion by Stas1981ru** [qutebrowser/qutebrowser#5777] [Q&A] [answered]
I have one machine with not fresh install of Qt and WebKit. I do not want to reinstall all system (Debian 10).
When I start QB I have error in file ./qutebrowser/browser/network/pac.py on line 30 :
`from PyQt5.QtQml import QJSEngine, QJSValue`
ImportError: /usr/local/lib/python3.7/dist-packages/PyQt5/QtQml.abi3.so: symbol _ZTI25QQmlEngineExtensionPlugin version Qt_5 not defined in file libQt5Qml.so.5 with link time reference
I think PyQt5 must be reinstalled.

Can U help to fast fix it ?

**Comment by The-Compiler** [comment]
What does `ldd /usr/local/lib/python3.7/dist-packages/PyQt5/QtQml.abi3.so` say?

**Reply by Stas1981ru** [reply]
linux-vdso.so.1 (0x00007ffc4b67a000)
	libQt5Gui.so.5 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libQt5Gui.so.5 (0x00007fe07af73000)
	libQt5Qml.so.5 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libQt5Qml.so.5 (0x00007fe07a8f6000)
	libQt5Core.so.5 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libQt5Core.so.5 (0x00007fe07a100000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fe079f70000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fe079daf000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fe079d8e000)
	libGL.so.1 => /lib/x86_64-linux-gnu/libGL.so.1 (0x00007fe079cf8000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fe079ada000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fe079957000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fe07993d000)
	libQt5Network.so.5 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libQt5Network.so.5 (0x00007fe079590000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fe079586000)
	libicui18n.so.56 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libicui18n.so.56 (0x00007fe0790eb000)
	libicuuc.so.56 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libicuuc.so.56 (0x00007fe078d33000)
	libicudata.so.56 => /usr/local/lib/python3.7/dist-packages/PyQt5/Qt/lib/libicudata.so.56 (0x00007fe077350000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fe07734b000)
	libgthread-2.0.so.0 => /lib/x86_64-linux-gnu/libgthread-2.0.so.0 (0x00007fe077346000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fe077227000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fe07bb73000)
	libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007fe0771f1000)
	libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007fe077134000)
	libgssapi_krb5.so.2 => /lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007fe0770e7000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fe077073000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007fe076f32000)
	libXext.so.6 => /lib/x86_64-linux-gnu/libXext.so.6 (0x00007fe076d1e000)
	libkrb5.so.3 => /lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007fe076c3e000)
	libk5crypto.so.3 => /lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007fe076c0a000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007fe076c04000)
	libkrb5support.so.0 => /lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007fe076bf5000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007fe076bee000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fe076bd2000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fe076ba8000)
	libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007fe0769a4000)
	libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fe07679e000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007fe076784000)

**Comment by The-Compiler** [comment, marked as answer]
Your `ldd` output looks like you did something like `sudo pip install PyQt5` at some point, so you're mixing the Debian Qt install with a system-wide PyQt install (which includes a pre-built Qt).

You should never `sudo pip install` anything, as it can mess up your system in various ways. You might want to `sudo pip uninstall PyQt5 PyQtWebEngine` and reinstall your system's Qt packages via `apt`. If you want a newer Qt for a specific project, use a virtualenv instead - for qutebrowser, see the relevant part of the [install instructions](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#tox).

**Reply by Stas1981ru** [reply]
Thanks, I try to uninstall and install Qt.

**Reply by Stas1981ru** [reply]
I made: pip uninstall PyQt5 PyQtWebEngine
All uninstalled ok.
To install I made:

apt install qt5-default
apt-get install python3-pyqt5
apt-get install pyqt5-dev-tools
apt-get install qttools5-dev-tools
apt-get install python3-pyqt5.qtwebengine

Now the folder /usr/local/lib/python3.7/dist-packages/PyQt5/Qt not contains any libs
So, then I try to start QB the error is: 
`File "./qutebrowser/misc/sql.py", line 25, in <module>
    from PyQt5.QtSql import QSqlDatabase, QSqlQuery, QSqlError
ModuleNotFoundError: No module named 'PyQt5.QtSql'`

Qt lib are installing via "pip3 install PyQtWebEngine" only (I did not find they install via APT)
With "pip3 install PyQtWebEngine" I still have the first error (mportError: /usr/local/lib/python3.7/dist-packages/PyQt5/QtQml.abi3.so: symbol _ZTI25QQmlEngineExtensionPlugin version Qt_5 not defined in file libQt5Qml.so.5 with link time reference).

**Reply by Stas1981ru** [reply]
I have copied /usr/local/lib/python3.7/dist-packages/PyQt5/ folder from other machine (QB work fine there) to this machine. So, ldd /usr/local/lib/python3.7/dist-packages/PyQt5/QtQml.abi3.so shows new valuse. But error ImportError: /usr/local/lib/python3.7/dist-packages/PyQt5/QtQml.abi3.so: symbol _ZTI25QQmlEngineExtensionPlugin version Qt_5 not defined in file libQt5Qml.so.5 with link time reference still exist.

**Reply by Stas1981ru** [reply]
Answer: the solution is "apt-get install python3-pyqt5.*"

**Reply by The-Compiler** [reply]
You won't need the entirety of PyQt, but you will need what's listed in the dependencies of the [qutebrowser package](https://packages.debian.org/buster/qutebrowser) (or just use that package instead?). In your case, I guess you were missing `python3-pyqt5.qtsql`.

**Reply by Stas1981ru** [reply]
>I guess you were missing `python3-pyqt5.qtsql` 
Yes, I try to add python3-pyqt5.xxx (where XXX were qtsql,qtqml,... ) and when add all (*)  )
Thanks for link to  qutebrowser package it can be better next time.
